src/main/msp/msp.c

```diff
@@ -1074,21 +1074,23 @@ static bool mspProcessOutCommand(int16_t cmdMSP, sbuf_t *dst)
 #if defined(USE_ACC)
             // Hack scale due to choice of units for sensor data in multiwii
 
-            uint8_t scale;
-            if (acc.dev.acc_1G > 512 * 4) {
+            float scale;
+            if(acc.dev.acc_1G == 2731){
+                scale = 16/3.0;
+            }else if (acc.dev.acc_1G > 512 * 4) {
                 scale = 8;
             } else if (acc.dev.acc_1G > 512 * 2) {
                 scale = 4;
             } else if (acc.dev.acc_1G >= 512) {
                 scale = 2;
-            } else {
+            }else {
                 scale = 1;
             }
 #endif
 
             for (int i = 0; i < 3; i++) {
 #if defined(USE_ACC)
-                sbufWriteU16(dst, lrintf(acc.accADC[i] / scale));
+                sbufWriteU16(dst, lrintf((int16_t)acc.accADC[i] / scale));
```

Comment:
Not sure what this is for. Can. you explain?

Reply:
this corssesponds to the change above, for getting the ACC scaling right in the configurator. In case that is not neededl, both changes are unnecessary.

Reply:
I am not sure I am following. acc.accADC is a float. The scale variable used to be uint8_t, but you changed it to float. Why do you need to convert accADC to integer before dividing it by the scale?

Reply:
I would understand if the (int16_t) was before the lrintf() function

Reply:
You are right, don't know why I introduced that cast. Doesn't make sense like that in the end.
I changed the scaling to float, since the scaling with +-12G is not an integer anymore.